impl_core/src/main/java/io/opencensus/implcore/tags/propagation/SerializationUtils.java

```diff
@@ -76,17 +77,19 @@ private SerializationUtils() {}
     // Use a ByteArrayDataOutput to avoid needing to handle IOExceptions.
     final ByteArrayDataOutput byteArrayDataOutput = ByteStreams.newDataOutput();
     byteArrayDataOutput.write(VERSION_ID);
+    int totalChars = 0; // Here chars are equivalent to bytes, since we're using ascii chars.
     for (Iterator<Tag> i = InternalUtils.getTags(tags); i.hasNext(); ) {
       Tag tag = i.next();
+      totalChars += tag.getKey().getName().length();
+      totalChars += tag.getValue().asString().length();
       encodeTag(tag, byteArrayDataOutput);
```

Comment:
Consider having encodeTag() return the number of characters it has encoded.

Reply:
Done.

Reply:
I don't have a strong opinion, but I think that the code was clearer before this change, because it was clearer that we were measuring the tags before they were serialized.

Reply:
I feel the same way, this also keeps the serialization and deserialization checking in sync. I prefer to use the previous logic.